### PR TITLE
Target as string or element (with tests)

### PR DESCRIPTION
**Pull request by orbweaver-**
PR for #571 (and PR #573)

This is building off of what @erickmonteiro did. I just added a couple tests and a warning.

I found that If you try to make a swal with a different target than your last swal, it just uses the last one, so I had to fix that before the tests would pass so I:
- Added a separate invalid string test and one valid and one invalid element tests.
- Made sure that if the target (string or element) is invalid, it is changed to "body"
- Made sure that if the target changes from the last modal, a new swal is made
- Made sure every time `dom.init` is called, it removes all the modals that we don't need.
- Removed the newly redundant warning in `dom.init` since it was replaced in `sweetalert2.setParameters`

I'm fairly certain this is what was needed, tell me if I missed something.

### CODE REVIEW

File: src/utils/dom.js [hunk 1]
```diff
@@ -15,6 +15,12 @@ export const states = {
  * Add modal + overlay to DOM
  */
 export const init = (params) => {
+  // Clean up old modals
+  while (getContainer()) {
```
Comment (limonte) [inline src/utils/dom.js]: There could be only one modal and one container at the time, this loop isn't needed I think.

Comment (orbweaver-) [inline src/utils/dom.js]: @limonte Yeah, that's where I was confused. I didn't see them being removed so I put that there. If I remove the lines now the tests break

Comment (limonte) [inline src/utils/dom.js]: > If I remove the lines now the tests break

For me tests are passing with this code:

```js
const c = getContainer()
c && c.parentNode.removeChild(c)
```

or

```js
const c = getContainer()
if (c) {
  c.parentNode.removeChild(c)
}
```

Which is basically the same.

Comment (orbweaver-) [inline src/utils/dom.js]: @limonte Ok, if it is doing it everytime then the loop is not needed. I'll update it

Comment (limonte) [inline src/utils/dom.js]: thanks :+:1:

Comment (orbweaver-) [inline src/utils/dom.js]: That should do it

**Comment by limonte** [comment]
[v6.6.8](https://github.com/limonte/sweetalert2/releases/tag/v6.6.8) :tada:

**Pull request merged**